client/src/components/ui/progress.tsx

```diff
@@ -0,0 +1,85 @@
+import Image from "next/image";
+import * as React from "react";
+
+export default function ProgressBar({ pageName }: { pageName: string }) {
+  const [progressWidth, setProgressWidth] = React.useState(0);
+  let progress = 0;
+  let titleName = "";
+
+  switch (pageName) {
+    case "schedule":
+      progress = 0;
+      titleName = "Drone Competition Schedule";
+      break;
+    case "format-rules":
+      progress = 30;
+      titleName = "Format & Rules";
+      break;
+    case "guests-sponsors":
+      titleName = "Guests & Sponsors";
+      progress = 60;
+      break;
+    case "leaderboard":
+      titleName = "Leaderboard";
+      progress = 80;
+      break;
+  }
+
+  React.useEffect(() => {
+    const updateProgress = () => {
+      if (pageName === "leaderboard") {
+        setProgressWidth(window.innerWidth >= 768 ? 90 : 80);
+      } else {
+        setProgressWidth(progress);
+      }
+    };
+
+    updateProgress();
+    window.addEventListener("resize", updateProgress);
+    return () => window.removeEventListener("resize", updateProgress);
+  }, [pageName, progress]);
+
+  return (
+    <div>
+      <div className="w-full flex-col justify-center">
+        <div className="mb-4 flex items-center justify-center">
+          <a className="title-large text-center text-[30px] text-dark md:text-[50px] lg:text-[60px]">
```

Comment:
You're using an `<a>` tag for a title, but it doesn't link anywhere. Use semantic HTML like `<h1>`.